File: crowdshield-ai/backend/app/schemas/alert.py
"""Alert schemas — alert feed, details, and response actions."""

import uuid
from datetime import datetime
from typing import Optional

from pydantic import BaseModel


class AlertResponse(BaseModel):
    id: uuid.UUID
    video_id: Optional[uuid.UUID] = None
    camera_id: Optional[uuid.UUID] = None
    level: str
    alert_type: str
    zone: Optional[str] = None
    reason: str
    confidence: float
    prediction_horizon: Optional[str] = None
    recommended_action: Optional[str] = None
    confirmation_frames: int
    persistence_seconds: float
    fused_confidence: float
    acknowledged: bool
    resolved: bool
    is_false_alarm: bool
    created_at: datetime

    model_config = {"from_attributes": True}


class AlertAcknowledge(BaseModel):
    acknowledged_by: str


class AlertListResponse(BaseModel):
    alerts: list[AlertResponse]
    total: int
    unacknowledged_count: int


class AlertStats(BaseModel):
    total: int
    info: int
    warning: int
    high: int
    critical: int
    false_alarm_rate: float
    avg_lead_time_seconds: float

File: crowdshield-ai/frontend/src/types/alert.ts
/* Types for alerts */

export type AlertLevel = 'info' | 'warning' | 'high' | 'critical';

export interface Alert {
  id: string;
  level: AlertLevel;
  alert_type: string;
  zone?: string;
  reason: string;
  confidence: number;
  prediction_horizon?: string;
  recommended_action?: string;
  confirmation_frames: number;
  persistence_seconds: number;
  fused_confidence: number;
  acknowledged: boolean;
  resolved: boolean;
  is_false_alarm: boolean;
  created_at: string;
}

export interface AlertStats {
  total: number;
  info: number;
  warning: number;
  high: number;
  critical: number;
  false_alarm_rate: number;
  avg_lead_time_seconds: number;
}
